src/engine/guidance/post_processing.cpp

```diff
@@ -538,19 +541,86 @@ std::vector<RouteStep> removeNoTurnInstructions(std::vector<RouteStep> steps)
 // that we come across.
 std::vector<RouteStep> postProcess(std::vector<RouteStep> steps)
 {
+#pragma message remove
     // the steps should always include the first/last step in form of a location
     BOOST_ASSERT(steps.size() >= 2);
     if (steps.size() == 2)
         return steps;
 
+    // Before we invalidate and remove silent instructions, we handle roundabouts (before they're
+    // getting collapsed into a single multi-hop instruction) by back-propagating exit lane
+    // constraints already to a roundabout's enter instruction.
+
+    const constexpr auto MIN_DURATION_NEEDED_FOR_LANE_CHANGE_INSIDE_ROUNDABOUT = 15.;
+
+    util::guidance::print(steps);
+
+    // Walk over all steps, find enter / leave roundabout pairs
+    for (auto it = begin(steps), last = end(steps); it != last;)
+    {
+        auto enter = std::find_if(it, last, [](const RouteStep &step) {
```

Comment:
this is unnecessarily large (O(n^2) instead of O(n)) complexity.
The enter should be the initial check. Only if *it is on enter, you need to process.